benchexec/tools/nitwit.py

```diff
@@ -48,30 +47,27 @@ def version(self, executable):
     def name(self):
         return "Nitwit"
 
-    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
-        if propertyfile:
-            options = options + ["-p", propertyfile]
-        return [executable] + options + tasks
+    def cmdline(self, executable, options, task, rlimits):
+        if task.property_file:
+            options = options + ["-p", task.property_file]
+        return [executable] + options + list(task.input_files_or_identifier)
 
-    def determine_result(self, returncode, returnsignal, output, isTimeout):
+    def determine_result(self, run):
         """
         See README.md at https://github.com/moves-rwth/nitwit-validator for information
         about result codes.
-        @param returncode: code returned
-        @param returnsignal: signal, which terminated validator
-        @param output: the std output
         @return: status of validator after executing a run
         """
-        if returnsignal == 0 and (returncode == 0 or returncode == 245):
+        if run.exit_code.signal is None and (run.exit_code.value == 0 or run.exit_code.value == 245):
             status = result.RESULT_FALSE_REACH
-        elif returncode is None or returncode in [-9, 9]:
+        elif run.exit_code.value is None or run.exit_code.value in [-9, 9]:
             status = "TIMEOUT"
-        elif returncode in [4, 5, 241, 242, 243, 250]:
+        elif run.exit_code.value in [4, 5, 241, 242, 243, 250]:
             status = result.RESULT_UNKNOWN
         else:
             status = result.RESULT_ERROR
 
-        if "Out of memory" in output or returncode == 251:
+        if run.output.any_line_contains("Out of memory") or (run.exit_code.value == 251):
```

Comment:
Here, the previous code was actually buggy and the condition `"Out of memory" in output` was never fulfilled because `output` contained only strings with an included line separator.

Is the intention to check that there is a line in the output that contains only `Out of memory` and nothing else except for the line separator? Then the condition would be `"Out of memory" in run.output` (the new API improves this use case and makes it possible to write this).

Or is the intention to check that the string appears somewhere in all of the output, even in the middle of some line somewhere? Then the new code is correct.

Reply:
Thanks for the clarification!